src/Features/Core/Portable/EditAndContinue/EditSession.cs

```diff
@@ -251,110 +237,139 @@ private LinePositionSpan GetUpToDateSpan(ActiveStatementDebugInfo activeStatemen
+                    // Can't calculate exception regions for active statements in out-of-sync documents.
+                    var (document, _) = await DebuggingSession.LastCommittedSolution.GetDocumentAndStateAsync(activeStatement.PrimaryDocumentId, cancellationToken).ConfigureAwait(false);
+                    if (document != null)
+                    {
+                        var sourceText = await document.GetTextAsync(cancellationToken).ConfigureAwait(false);
+                        var syntaxRoot = await document.GetSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
 
-                    var analyzer = document.Project.LanguageServices.GetService<IEditAndContinueAnalyzer>();
-                    var exceptionRegions = analyzer.GetExceptionRegions(sourceText, syntaxRoot, activeStatement.Span, activeStatement.IsNonLeaf, out var isCovered);
+                        var analyzer = document.Project.LanguageServices.GetService<IEditAndContinueAnalyzer>();
+                        exceptionRegions = analyzer.GetExceptionRegions(sourceText, syntaxRoot, activeStatement.Span, activeStatement.IsNonLeaf, out isCovered);
+                    }
+                    else
+                    {
+                        // Document is either out-of-sync, design-time-only or missing from the baseline.
+                        // If it's missing or design-time-only it can't have active statements.
+                        hasOutOfSyncDocuments = true;
+                        isCovered = false;
+                        exceptionRegions = default;
+                    }
 
                     builder[activeStatement.Ordinal] = new ActiveStatementExceptionRegions(exceptionRegions, isCovered);
                 }
 
-                return builder.ToImmutable();
+                var result = builder.ToImmutable();
+
+                // Only cache results if no active statements are in out-of-sync documents.
+                if (!hasOutOfSyncDocuments)
+                {
+                    ImmutableInterlocked.InterlockedInitialize(ref _lazyBaseActiveExceptionRegions, result);
+                }
+
+                return result;
             }
             catch (Exception e) when (FatalError.ReportWithoutCrashUnlessCanceled(e))
             {
                 return ImmutableArray<ActiveStatementExceptionRegions>.Empty;
             }
         }
 
-        private ImmutableArray<(DocumentId DocumentId, AsyncLazy<DocumentAnalysisResults> Results)> GetChangedDocumentsAnalyses(Project baseProject, Project project)
+        private async Task<(ImmutableArray<(Document Document, AsyncLazy<DocumentAnalysisResults> Results)>, ImmutableArray<Diagnostic> Diagnostics)> GetChangedDocumentsAnalysesAsync(
+            Project baseProject, Project project, CancellationToken cancellationToken)
         {
+            var changedDocuments = ArrayBuilder<(Document Old, Document New)>.GetInstance();
```

Comment:
Don't we do camelCase for tuple members?

Reply:
I don't think there is a common convention.